include/frepple/model.h

```diff
@@ -2585,6 +2590,9 @@ class OperationPlan : public Object,
   /* Quantity. */
   double quantity = 0.0;
 
+  /* Demand name to identify which demand takes this operationplan */
+  string demandReference = "";
```

Comment:
Can't you use the existing batch field for this? 
Operationplan is one of the biggest memory consumers in a big modeL  I'll be very hesitant to merge a pull a request that adds a string to every instance.